.github/workflows/actions_build.yml

```diff
@@ -26,53 +27,44 @@ jobs:
       fail-fast: false
       matrix:
         on: [ self-hosted, macos-12, windows-latest ]
-        java: [ 17 ]
+        java: [ 19 ]
         include:
           - java: 8
             on: self-hosted
           - java: 11
             on: self-hosted
-          - java: 16
+          - java: 17
             on: self-hosted
             leak: true
-          - java: 17
+          - java: 19
             # TODO(ikhoon): Revert to self-hosted runners once the following error is fixed
             #               `Cannot expand ZIP '/actions-runner/../armeria-shaded-1.7.3-SNAPSHOT.jar' as it does not exist.`
             on: macos-12
             coverage: true
-          - java: 17
+          - java: 19
             on: self-hosted
             snapshot: true
 
     steps:
       - uses: actions/checkout@v2
 
       - id: setup-jdk
-        if: ${{ matrix.java != 17 }}
+        if: ${{ matrix.java != env.BUILD_JDK_VERSION }}
         name: Set up JDK ${{ matrix.java }}
         uses: actions/setup-java@v2
         with:
           distribution: 'temurin'
           java-version: ${{ matrix.java }}
 
-      - id: setup-jdk-17
-        name: Set up JDK 17
+      - id: setup-build-jdk
+        name: Set up build JDK ${{ env.BUILD_JDK_VERSION }}
         uses: actions/setup-java@v2
         with:
           distribution: 'temurin'
-          java-version: '17'
+          java-version: ${{ env.BUILD_JDK_VERSION }}
 
-      - if: ${{ matrix.on != 'self-hosted' }}
-        name: Restore the cache
-        uses: actions/cache@v2
-        with:
-          path: |
-            ~/.gradle/wrapper
-            ~/.gradle/caches
-          key: build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-${{ hashFiles('gradle.properties', 'gradle/wrapper/gradle-wrapper.properties', '**/build.gradle', 'dependencies.yml', '*/package-lock.json') }}
-          restore-keys: |
-            build-${{ matrix.java }}-${{ runner.os }}-${{ secrets.CACHE_VERSION }}-
-            build-${{ matrix.java }}-${{ runner.os }}-
+      - name: Setup Gradle
+        uses: gradle/gradle-build-action@v2
```

Comment:
So it works now? 😄

Reply:
I checked the GitHub Actions log and it does not seem to work.
Both the original one and the new one also didn't work on GitHub-managed hosts.
- Original
<img width="824" alt="image" src="https://user-images.githubusercontent.com/1866157/208369798-4a47da8f-2ba7-4ab6-a9f3-f9cc1bac9e99.png">

- gradle-build-actions
<img width="708" alt="image" src="https://user-images.githubusercontent.com/1866157/208373324-f3ff9ced-7123-4489-aab7-a3d7b15c4de2.png">

Reply:
It seems to work now. There might be some complicate conditions to use cache. 
<img width="731" alt="image" src="https://user-images.githubusercontent.com/1866157/209598322-af454534-4d69-43e9-871c-5babb76a0fcd.png">